interchain-token-service/src/contract/execute.rs

```diff
@@ -0,0 +1,291 @@
+use cosmwasm_std::{DepsMut, HexBinary, Response};
+use error_stack::{report, Result, ResultExt};
+use router_api::{Address, ChainName, CrossChainId};
+
+use crate::contract::Error;
+use crate::events::ItsContractEvent;
+use crate::primitives::ItsHubMessage;
+use crate::state::{self, load_config, load_its_address};
+
+/// Executes an incoming ITS message.
+///
+/// This function handles the execution of ITS (Interchain Token Service) messages received from
+/// its sources. It verifies the source address, decodes the message, applies balance tracking,
+/// and forwards the message to the destination chain.
+pub fn execute_message(
+    deps: DepsMut,
+    cc_id: CrossChainId,
+    source_address: Address,
+    payload: HexBinary,
+) -> Result<Response, Error> {
+    let config = load_config(deps.storage).change_context(Error::InvalidStoreAccess)?;
+
+    let source_chain = ChainName::try_from(cc_id.source_chain.clone().to_string())
+        .change_context(Error::InvalidPayload)?;
```

Comment:
Isn't this going to error if source chain is a legacy chain with uppercase letters?